pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -75,6 +75,8 @@ type Values struct {
 	Workers []gardencorev1beta1.Worker
 	// KubernetesVersion is the Kubernetes version of the cluster for which the worker nodes shall be created.
 	KubernetesVersion *semver.Version
+	// CloudProfile is the CloudProfile referenced by the shoot
+	CloudProfile *gardencorev1beta1.CloudProfile
```

Comment:
Nit: Instead of passing the entire `CloudProfile`, what about only passing the `MachineTypes`?